File: py/99.py
from math import log

with open('../input/99') as f:
    pairs = f.read().split('\n')[0:-1]
    pairs = map(lambda p: map(int, p.split(',')), pairs)
    pairs = map(lambda p: log(p[0]) * p[1], pairs)

    print pairs.index(max(pairs)) + 1

File: clj/99.clj
(def nums (map #(read-string (apply str "[" % "]"))
               (re-seq #"\S+" (slurp "../input/99"))))

(def zipped (zipmap (range 1 1001) nums))

(prn (first
       (apply max-key
              #(* (Math/log (first (second %)))
                  (second (second %)))
              zipped)))
